packages/babel-parser/src/parser/statement.js

```diff
@@ -1666,6 +1680,33 @@ export default class StatementParser extends ExpressionParser {
     );
   }
 
+  pushClassAccessorProperty(
+    classBody: N.ClassBody,
+    prop: N.ClassAccessorProperty,
+    isPrivate: boolean,
+  ) {
+    if (
+      !prop.computed &&
+      prop.key.type !== "PrivateName" &&
```

Comment:
When the `["estree", { "classFeatures": true }]` plugin is enabled, the type of private names is diferent. I think we have a `this.isPrivateName(prop.key)` (or something similar) method to handle both the AST shapes.

Reply:
```suggestion
      !isPrivate &&
```

Reply:
I don't think that `isPrivate` does the type guarding correctly, which is the issue here, so going to go with `this.isPrivateName` to get that assertion

Reply:
actually it is more performant not to do the check probably, so I can just use type casting

Reply:
Yeah you can perfectly ignore the FlowError `// $FlowIgnore` or use `/*:: invariant(prop.key.type !== "PrivateName") */`. Flow does not understand the equivalence between the token type and AST node type anyway.